comictaggerlib/ctsettings/__init__.py

```diff
@@ -25,3 +34,106 @@
     "ct_ns",
     "group_for_plugin",
 ]
+
+
+class SettingsEncoder(json.JSONEncoder):
+    def default(self, obj: Any) -> Any:
+        if isinstance(obj, pathlib.Path):
+            return str(obj)
+
+        # Let the base class default method raise the TypeError
+        return json.JSONEncoder.default(self, obj)
+
+
+def validate_types(config: settngs.Config[settngs.Values]) -> settngs.Config[settngs.Values]:
+    # Go through each setting
+    for group in config.definitions.values():
+        for setting in group.v.values():
+            # Get the value and if it is the default
+            value, default = settngs.get_option(config.values, setting)
+            if not default:
+                if setting.type is not None:
+                    # If it is not the default and the type attribute is not None
+                    # use it to convert the loaded string into the expected value
+                    if isinstance(value, str):
+                        config.values[setting.group][setting.dest] = setting.type(value)
+    return config
+
+
+def parse_file(definitions: settngs.Definitions, filename: pathlib.Path) -> tuple[settngs.Config[settngs.Values], bool]:
+    """
+    Read options from a json dictionary from a file.
+
+    Args:
+        definitions: A set of setting definitions. See `Config.definitions` and `Manager.definitions`
+        filename: A pathlib.Path object to read a json dictionary from
+    """
+    options: settngs.Values = {}
+    success = True
+    if filename.exists():
+        try:
+            with filename.open() as file:
+                opts = json.load(file)
+            if isinstance(opts, dict):
+                options = opts
+            else:  # pragma: no cover
+                raise Exception("Loaded file is not a JSON Dictionary")
+        except Exception:  # pragma: no cover
+            logger.exception("Failed to load config file: %s", filename)
+            success = False
+    else:
+        logger.info("No config file found")
+        success = True
+
+    return settngs.normalize_config(validate_types(settngs.Config(options, definitions)), file=True), success
```

Comment:
Is there a reason to not use the function from settngs? validate_types does not need to happen in this function.

Reply:
Nope. Removed and squished.